rlp/sedes/serializable.py

```diff
@@ -354,14 +366,20 @@ def __new__(cls, name, bases, attrs):
             in zip(meta.field_names, meta.field_attrs)
         }
 
+        removed_parent_field_props = {
+            field_name: RemovedParentField(field_name)
+            for base in bases if issubclass(base, Serializable) if hasattr(base, '_meta')
+            for field_name in base._meta.field_names if field_names not in field_props
+        }
+
         return super_new(
             cls,
             name,
             bases,
             dict(
                 tuple(field_props.items()) +
-                tuple(attrs.items()) +
-                (('__slots__', meta.field_attrs),)
+                tuple(removed_parent_field_props.items()) +
```

Comment:
Maybe this can be simplified with:

```py
        removed_parent_field_props = tuple(
            (field_name, RemovedParentField(field_name))
            for base in bases if issubclass(base, Serializable) if hasattr(base, '_meta')
            for field_name in base._meta.field_names if field_names not in field_props
        )
...
             dict(
                 field_props +
                 removed_parent_field_props +
```

---

Is there really no way to just prevent the attribute from being created in the child in the first place? That seems like the cleanest solution. But, I'm new to metaclasses, so no specific suggestions.

Reply:
This is the best I could come up with.  It's not really a metaclass thing but rather a basic object inheritance thing.  

Child classes natively inherit everything their parents had and so we'd either have to do something silly like dynamically creating a new parent class with all of the field attributes stripped off to inherit from and then using `ABCMeta.register` to ensure things like `isinstance` and `issubclass` still returned accurately.

I didn't actually try doing this because I was pretty confident that it'd be fragile and have some weird side effects that I couldn't predict up front.

:(

Reply:
I just thought of an alternative which seem obvious now having thought of it.

I'm going to change this so that it's invalid to subclass a `Serializable` if the subclass doesn't contain a superset of the parent classes fields.  Makes wayy more sense to me and remove this nonsense.